### ISSUE TITLE
podspec is missing 'Network' dependency

### ISSUE THREAD

**Issue by SpacyRicochet** [rwbutler/Connectivity#24]
Connectivity has a dependency on the `Network` framework; apps built with Connectivity – without being linked against `Network` themselves – will crash when Connectivity calls one of `Network`'s methods.

By [adding the dependency to the podspec](https://guides.cocoapods.org/syntax/podspec.html), CocoaPods should be able to infer this themselves.

```
spec.framework      = 'Network'
```

This avoids having to link against `Network` manually.

**Comment by rwbutler** [comment]
Thanks for raising this issue 👍🏻I've made this change in 3.1.1 - hopefully this will resolve the issue.

**Comment by Alond** [comment]
There trouble with 3.1.1
`dyld: Library not loaded: /System/Library/Frameworks/Network.framework/Network
  Referenced from: /private/var/containers/Bundle/Application/2C7BAC9F-0A16-496F-BEEB-5ACDD33FBD97/CarrotApp.app/Frameworks/Connectivity.framework/Connectivity
  Reason: image not found
(lldb) `

3.1.0 works fine

**Comment by rwbutler** [comment]
Thanks for letting me know - I'll check into this.

**Comment by Alond** [comment]
If it's helps trouble on device with iOS 11.2.6, on device with iOS 12.4 it's ok

**Comment by rlasante** [comment]
I'm seeing the 
`
dyld: Library not loaded: /System/Library/Frameworks/Network.framework/Network Referenced from: /private/var/containers/Bundle/Application/2C7BAC9F-0A16-496F-BEEB-5ACDD33FBD97/CarrotApp.app/Frameworks/Connectivity.framework/Connectivity Reason: image not found (lldb)
`

 too when building with Xcode11GM2, I was able to resolve the issue by making the `Network.framework` optional in the Pod's target but I'm running into issues when attempting to build unit tests when targeting iOS 12 or iOS 13

`
dyld: Library not loaded: /System/Library/PrivateFrameworks/Network.framework/Network
  Referenced from: /Users/ryan.lasante/Library/Developer/XCTestDevices/E7A4D0B6-9C27-411F-B67F-4BE4001EB13C/data/Containers/Bundle/Application/96353817-7C0C-4985-9508-ECE4F11A44FA/Automated Documentation.app/Automated Documentation
  Reason: no suitable image found.  Did find:
	/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 12.0.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/Network.framework/Network: mach-o, but not built for iOS simulator
	/System/Library/PrivateFrameworks/Network.framework/Network: mach-o, but not built for iOS simulator
`

as said above, not an issue when using 3.1.0

**Comment by rwbutler** [comment]
Just a quick update - I started investigating this today. Creating a brand new Xcode project, I integrated Connectivity 3.1.1 and built to an iPhone 8 and iPhone X both running iOS 13.0, an iPad Mini 4 running iOS 11.2.2 and iPhone 6S running iOS 11.4 but unfortunately wasn't able to reproduce this issue.

Here's the code I used to try and reproduce the issue:
https://github.com/rwbutler/connectivitytest

If there are any obvious distinctions between this project and your own that might help track down the issue let me know and I'll try to pursue this further.
